code/modules/mob/living/basic/heretic/maid_in_the_mirror.dm

```diff
@@ -16,36 +16,34 @@
 	melee_damage_upper = 16
 	sight = SEE_MOBS | SEE_OBJS | SEE_TURFS
 	death_message = "shatters and vanishes, releasing a gust of cold air."
-	loot = list(
-		/obj/item/shard,
+	/// Whether we take damage when someone looks at us
+	var/harmed_by_examine = TRUE
+	/// How often being examined by a specific mob can hurt us
+	var/recent_examine_damage_cooldown = 10 SECONDS
+	/// A list of REFs to people who recently examined us
+	var/list/recent_examiner_refs = list()
+
+/mob/living/basic/heretic_summon/Initialize(mapload)
+	. = ..()
+	var/static/list/loot = list(
 		/obj/effect/decal/cleanable/ash,
 		/obj/item/clothing/suit/armor/vest,
 		/obj/item/organ/internal/lungs,
+		/obj/item/shard,
 	)
-	actions_to_add = list(/datum/action/cooldown/spell/jaunt/mirror_walk)
+	AddElement(/datum/element/death_drops, loot)
+	var/datum/action/cooldown/spell/jaunt/mirror_walk/jaunt = new (src)
+	jaunt.Grant(src)
 
-	/// Whether we take damage when we're examined
-	var/weak_on_examine = TRUE
-	/// The cooldown after being examined that the same mob cannot trigger it again
-	var/recent_examine_damage_cooldown = 10 SECONDS
-	/// A list of REFs to people who recently examined us
-	var/list/recent_examiner_refs = list()
-
-/mob/living/simple_animal/hostile/heretic_summon/maid_in_the_mirror/death(gibbed)
+/mob/living/basic/heretic_summon/maid_in_the_mirror/death(gibbed)
 	var/turf/death_turf = get_turf(src)
-	death_turf.TakeTemperature(-40)
+	death_turf.TakeTemperature(-40) // Spooky
 	return ..()
 
 // Examining them will harm them, on a cooldown.
-/mob/living/simple_animal/hostile/heretic_summon/maid_in_the_mirror/examine(mob/user)
+/mob/living/basic/heretic_summon/maid_in_the_mirror/examine(mob/user)
 	. = ..()
-	if(!weak_on_examine)
-		return
-
-	if(!isliving(user) || user.stat == DEAD)
-		return
-
-	if(IS_HERETIC_OR_MONSTER(user) || user == src)
+	if(!harmed_by_examine || !isliving(user) || user.stat == DEAD || IS_HERETIC_OR_MONSTER(user) || user == src)
```

Comment:
lets reorganize this to short circuit better (keeping the more expensive stuff near the end)

```suggestion
	if(!harmed_by_examine || user == src || user.stat == DEAD || !isliving(user) || IS_HERETIC_OR_MONSTER(user))
```

i'm pretty sure it doesn't matter in the long run but lets be safe in case someone decides to spam examine or something